spec/mailers/user_mailer_spec.rb

```diff
@@ -60,4 +60,122 @@
       expect(mail.body).to include(reset_password_internal_user_url(user, token: user.reset_password_token))
     end
   end
+
+  describe '#got_new_comment' do
+    let(:user) { create(:learner) }
+    let(:token) { AuthenticationToken.find_by(user: user) }
+    let(:request_for_comment) { create(:rfc_with_comment, user: user) }
+    let(:commenting_user) { InternalUser.create(attributes_for(:teacher)) }
+    let(:mail) { described_class.got_new_comment(request_for_comment.comments.first, request_for_comment, commenting_user).deliver_now }
+
+    it 'sets the correct sender' do
+      expect(mail.from).to include(CodeOcean::Application.config.action_mailer[:default_options][:from])
+    end
+
+    it 'sets the correct subject' do
+      expect(mail.subject).to eq(I18n.t('mailers.user_mailer.got_new_comment.subject', commenting_user_displayname: commenting_user.displayname))
+    end
+
+    it 'sets the correct receiver' do
+      expect(mail.to).to include(request_for_comment.user.email)
+    end
+
+    it 'includes the correct URL' do
+      expect(mail.body).to include(request_for_comment_url(request_for_comment))
+    end
+
+    it 'includes the correct authentication token' do
+      expect(mail.body).to include(token.shared_secret)
+    end
+
+    it 'creates a new authentication token' do
+      expect { mail }.to change(AuthenticationToken, :count).by(1)
+    end
+
+    it 'sets a non-expired authentication token' do
```

Comment:
What is this supposed to be testing? Especially the second part (lines 98-99) looks suspicious to me.

Reply:
We removed the two suspicious lines and improved the spec. It now checks that a token valid for seven days is created for authentication.